kyuubi-server/src/main/scala/org/apache/kyuubi/engine/spark/SparkProcessBuilder.scala

```diff
@@ -259,6 +260,15 @@ class SparkProcessBuilder(
     map.result().toMap
   }
 
+  def extraYarnConf: Map[String, String] = {
+    if (clusterManager().exists(_.toLowerCase(Locale.ROOT).startsWith("yarn"))) {
+      // set `spark.yarn.maxAppAttempts` to 1 to avoid invalid attempts.
+      Map(YARN_MAX_APP_ATTEMPTS_KEY -> "1")
```

Comment:
[YARN-5617](https://issues.apache.org/jira/browse/YARN-5617). AMs only intended to run one attempt can be run more than once


Fix Version/s: 2.9.0, 3.0.0-alpha4, 2.8.2

Reply:
maybe we should add such info to the comment